File: Selenium/Activities/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:

    wait = WebDriverWait(driver, 10)

    driver.get("https://training-support.net/webelements/dynamic-controls")

    print("Page title:", driver.title)

    checkbox = driver.find_element(By.ID, "checkbox")
    print("Checkbox visible:", checkbox.is_displayed())

    driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']").click()

    wait.until(EC.invisibility_of_element_located((By.ID, "checkbox")))

    print("Checkbox visible after removal: False")

    driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']").click()

    wait.until(EC.visibility_of_element_located((By.ID, "checkbox")))

    checkbox = driver.find_element(By.ID, "checkbox")
    checkbox.click()

    print("Checkbox selected:", checkbox.is_selected())

File: Selenium/Activity8.java
package examples;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity8 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-controls");

        System.out.println("Page title: " + driver.getTitle());

        System.out.println("Checkbox visible: " +
                driver.findElement(By.id("checkbox")).isDisplayed());

        driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();

        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id("checkbox")));

        System.out.println("Checkbox visible after removal: false");

        driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("checkbox")));

        driver.findElement(By.id("checkbox")).click();

        System.out.println("Checkbox selected: " +
                driver.findElement(By.id("checkbox")).isSelected());

        driver.quit();
    }
}

